diff --git a/_cjs/core/internal/webauthn.js b/_cjs/core/internal/webauthn.js
index 242460ecbcc876b35a1f63b579755999d8bde51e..4f750837135e0f973c2ebd80db5ca6bee6543368 100644
--- a/_cjs/core/internal/webauthn.js
+++ b/_cjs/core/internal/webauthn.js
@@ -32,7 +32,7 @@ async function parseCredentialPublicKey(response) {
         return PublicKey.from(publicKey);
     }
     catch (error) {
-        if (error.message !== 'Permission denied to access object')
+      if (error.message !== 'Permission denied to access object' && !error.message.includes("spki") && !error.message.includes('undefined'))
             throw error;
         const data = new Uint8Array(response.attestationObject);
         const coordinateLength = 0x20;
diff --git a/_esm/core/internal/webauthn.js b/_esm/core/internal/webauthn.js
index 5620414096df11145e4c92d0fd2616690c1b1580..891f5f31518c61c520edb401eec47c052a1a1f45 100644
--- a/_esm/core/internal/webauthn.js
+++ b/_esm/core/internal/webauthn.js
@@ -43,7 +43,7 @@ export async function parseCredentialPublicKey(response) {
         // Fallback for 1Password Firefox Add-on restricts access to certain credential properties
         // so we need to use `attestationObject` to extract the public key.
         // https://github.com/passwordless-id/webauthn/issues/50#issuecomment-2072902094
-        if (error.message !== 'Permission denied to access object')
+        if (error.message !== 'Permission denied to access object' && !error.message.includes("spki"))
             throw error;
         const data = new Uint8Array(response.attestationObject);
         const coordinateLength = 0x20;
